src/OrbitVulkanLayer/SubmissionTrackerTest.cpp

```diff
@@ -536,6 +538,99 @@ TEST_F(SubmissionTrackerTest,
+      .WillRepeatedly(Invoke(mock_enqueue_capture_event));
+
+  producer_->StartCapture();
+
+  // VkCommandBuffer's are opaque handles and we can't allocate them properly as we can't create
+  // a device in the context of the text. Using absl::bit_cast is a workaround to get two
+  // VkCommandBuffer's that are different, which is needed for this test.
+  std::array<VkCommandBuffer, 2> command_buffers{absl::bit_cast<VkCommandBuffer>(1L),
+                                                 absl::bit_cast<VkCommandBuffer>(2L)};
+  CHECK(command_buffers[0] != command_buffers[1]);
+
+  tracker_.TrackCommandBuffers(device_, command_pool_, &command_buffers[0], 2);
+  tracker_.MarkCommandBufferBegin(command_buffers[0]);
+  tracker_.MarkCommandBufferEnd(command_buffers[0]);
+  tracker_.MarkCommandBufferBegin(command_buffers[1]);
+  tracker_.MarkCommandBufferEnd(command_buffers[1]);
+
+  std::array<VkSubmitInfo, 2> submit_infos{VkSubmitInfo{.sType = VK_STRUCTURE_TYPE_SUBMIT_INFO,
+                                                        .pNext = nullptr,
+                                                        .pCommandBuffers = &command_buffers[0],
+                                                        .commandBufferCount = 1},
+                                           VkSubmitInfo{.sType = VK_STRUCTURE_TYPE_SUBMIT_INFO,
+                                                        .pNext = nullptr,
+                                                        .pCommandBuffers = &command_buffers[1],
+                                                        .commandBufferCount = 1}};
+
+  pid_t pid = orbit_base::GetCurrentThreadId();
+  std::array<uint64_t, 2> pre_submit_times;
+  std::array<uint64_t, 2> post_submit_times;
+
+  pre_submit_times[0] = orbit_base::CaptureTimestampNs();
+  std::optional<QueueSubmission> queue_submission_optional =
+      tracker_.PersistCommandBuffersOnSubmit(queue_, 1, &submit_infos[0]);
+  tracker_.PersistDebugMarkersOnSubmit(queue_, 1, &submit_infos[0], queue_submission_optional);
+  post_submit_times[0] = orbit_base::CaptureTimestampNs();
+
+  pre_submit_times[1] = orbit_base::CaptureTimestampNs();
+  std::optional<QueueSubmission> queue_submission_optional2 =
+      tracker_.PersistCommandBuffersOnSubmit(queue_, 1, &submit_infos[1]);
+  tracker_.PersistDebugMarkersOnSubmit(queue_, 1, &submit_infos[1], queue_submission_optional2);
+  post_submit_times[1] = orbit_base::CaptureTimestampNs();
+
+  tracker_.CompleteSubmits(device_);
+  tracker_.CompleteSubmits(device_);
+
+  EXPECT_THAT(actual_reset_slots1, UnorderedElementsAre(kSlotIndex1, kSlotIndex2, kSlotIndex4));
+  // We failed on the 4th GetQuerySlots call above, which belongs to the "begin" timestamp of the
+  // second command buffer (note that the order of calls when querying for timestamps is to query
+  // for the end timestamp first). Therefore, this must be kSlotIndex3.
+  // TODO: Consider if we should just check the size of the reset slots? Depending on code order
```

Comment:
Thoughts?

Reply:
I was also thinking about this. I guess it is fine for the tests. We had the tests from quite early on, and changed the code and the tests quite a bit. AFAIR we did't had issues with slots indices. So I think this is fine for the tests. Alternative would be to collect the slots into the same vector, and just check that all of them are reset exactly once.

Reply:
Ok, if we find it bothers us, we can change it later. I'll merge as-is. Thanks for the input!